internal/xds/bootstrap/bootstrap.yaml.tpl

```diff
@@ -125,6 +125,11 @@ static_resources:
           http2_protocol_options: {}
     name: xds_cluster
     type: STRICT_DNS
+    upstream_connection_options:
+      tcp_keepalive:
+        keepalive_probes: 3
```

Comment:
where are these number came from?

Reply:
I used the same ones as Contour, but we can use anything really

Reply:
it's OK now, just want to clearify.

Reply:
since we control the server (xds server https://github.com/envoyproxy/gateway/blob/72bfe9bde95015b6731857f52ea30d27ee85145d/internal/xds/server/runner/runner.go#L94C12-L94C22 &  https://pkg.go.dev/net#ListenConfig) and client (this envoy cluster), we can use higher values to reduce cpu ?

Reply:
how does controlling the server help? the goal is for the client detect half closed situation fairly quickly. i think the server will detect this anyway whenever it sends an update. i dont feel strongly about what values we use though, if you have a preference i am all ears!

Reply:
i think either is fine personally

Reply:
here are the comments in the `bootstrap.yaml` in https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#aggregated-discovery-service
```
   # It is recommended to configure either HTTP/2 or TCP keepalives in order to detect
    # connection issues, and allow Envoy to reconnect. TCP keepalive is less expensive, but
    # may be inadequate if there is a TCP proxy between Envoy and the management server.
    # HTTP/2 keepalive is slightly more expensive, but may detect issues through more types
    # of intermediate proxies.
```

so boils down to, if we believe there might ever be an proxy b/w the xds server and client (like a sidecar :) ) , it might make more sense to stick with HTTP/2 keepalives

Reply:
looks like even contour eventually moved to HTTP2 keep alives eventually https://github.com/projectcontour/contour/issues/3882

Reply:
ok, i can do this :)

Reply:
done

Reply:
awesome thanks for doing this in the client/envoy, this will also need to be done in the peer/server
```
grpc.KeepaliveParams
```

https://pkg.go.dev/google.golang.org/grpc/keepalive#ServerParameters

Reply:
This is for server detecting client going away, right? I feel its a separate discussion, I suppose the goal would be saving some resources on server side, but it won't matter for clients, right?

Reply:
for grpc keep alive the server needs to be set up with similar grpc keep alive settings, else the client might receive `GO_AWAY ENHANCE_YOUR_CALM`
https://grpc.io/docs/guides/keepalive/#background

Reply:
done :)